plugins/obs-filters/noise-suppress-filter.c

```diff
@@ -257,14 +390,97 @@ static void noise_suppress_update(void *data, obs_data_t *s)
 		ng->rnn_resampler_back = audio_resampler_create(&src, &dst);
 	}
 #endif
+#ifdef LIBNVAFX_ENABLED
+	if (sample_rate == NVAFX_SAMPLE_RATE) {
+		ng->nvafx_resampler = NULL;
+		ng->nvafx_resampler_back = NULL;
+	} else {
+		struct resample_info src, dst;
+		src.samples_per_sec = sample_rate;
+		src.format = AUDIO_FORMAT_FLOAT_PLANAR;
+		src.speakers = convert_speaker_layout((uint8_t)channels);
+
+		dst.samples_per_sec = NVAFX_SAMPLE_RATE;
+		dst.format = AUDIO_FORMAT_FLOAT_PLANAR;
+		dst.speakers = convert_speaker_layout((uint8_t)channels);
+
+		ng->nvafx_resampler = audio_resampler_create(&dst, &src);
+		ng->nvafx_resampler_back = audio_resampler_create(&src, &dst);
+	}
+#endif
 }
 
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4706)
+#endif
+bool load_nvafx(void)
+{
+#ifdef LIBNVAFX_ENABLED
+	if (!load_lib()) {
+		blog(LOG_INFO,
+		     "[noise suppress: Nvidia RTX denoiser disabled, redistributable not found]");
+		return false;
+	} else {
+		blog(LOG_INFO, "[noise suppress: Nvidia RTX denoiser enabled]");
+	}
+
+#define LOAD_SYM_FROM_LIB(sym, lib, dll)                                   \
+	if (!(sym = (sym##_t)GetProcAddress(lib, #sym))) {                 \
+		DWORD err = GetLastError();                                \
+		printf("[noise suppress: Couldn't load " #sym " from " dll \
+		       ": %lu (0x%lx)]",                                   \
+		       err, err);                                          \
+		goto unload_everything;                                    \
+	}
+
+#define LOAD_SYM(sym) LOAD_SYM_FROM_LIB(sym, nv_audiofx, "NVAudioEffects.dll")
+	LOAD_SYM(NvAFX_GetEffectList);
+	LOAD_SYM(NvAFX_CreateEffect);
+	LOAD_SYM(NvAFX_DestroyEffect);
+	LOAD_SYM(NvAFX_SetU32);
+	LOAD_SYM(NvAFX_SetString);
+	LOAD_SYM(NvAFX_SetFloat);
+	LOAD_SYM(NvAFX_GetU32);
+	LOAD_SYM(NvAFX_GetString);
+	LOAD_SYM(NvAFX_GetFloat);
+	LOAD_SYM(NvAFX_Load);
+	LOAD_SYM(NvAFX_Run);
+#undef LOAD_SYM
+	nvafx_loaded = true;
+	return true;
+
+unload_everything:
+	release_lib();
+#endif
+	return false;
+}
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
 static void *noise_suppress_create(obs_data_t *settings, obs_source_t *filter)
 {
 	struct noise_suppress_data *ng =
 		bzalloc(sizeof(struct noise_suppress_data));
 
 	ng->context = filter;
+
+#ifdef LIBNVAFX_ENABLED
+	char sdk_path[MAX_PATH];
+	nvafx_get_sdk_path(sdk_path);
+	if (!sdk_path) {
+		ng->use_nvafx = false;
+	} else {
+		const char *file = "\\models\\denoiser_48k.trtpkg";
+		ng->sdk_path = sdk_path;
+		int size = strlen(sdk_path) + strlen(file) + 1;
+		char *buffer = (char *)bmalloc(size);
+		strcpy(buffer, sdk_path);
+		strcat(buffer, file);
+		ng->model = buffer;
+	}
+#endif
```

Comment:
Again this is another one of those cases where the code feels a bit squishier than I'd prefer. It's just a nitpick, but I feel like it could look a bit cleaner. Like:

```c
	char sdk_path[MAX_PATH];
	nvafx_get_sdk_path(sdk_path);

	if (!sdk_path) {
		ng->use_nvafx = false;
	} else {
		const char *file = "\\models\\denoiser_48k.trtpkg";
		size_t size = strlen(sdk_path) + strlen(file) + 1;
		char *buffer = (char *)bmalloc(size);

		ng->sdk_path = sdk_path;
		strcpy(buffer, sdk_path);
		strcat(buffer, file);
		ng->model = buffer;
	}
```

Additionally, `strlen()` returns a `size_t` value, not an `int` value.

Reply:
thanks, fixed